### DISCUSSION TITLE
Make custom-filetypes more powerful?

### DISCUSSION THREAD

**Discussion by ghost** [geany/geany#3669] [General]
For example, being able to overwrite the comment syntax of the Scintilla lexer: the language is a C-style language but uses Ruby's syntax for comment: `#` instead of `//` for line comment and `=begin` and `=end` instead of `/*` and `*/` for block comment. Currently, this is impossible. The answer will always be go and add this to Scintilla. But, is this really needed? Notepad++ is also using Scintilla and as I know it can do this.

**Comment by elextr** [comment]
Lexers are provided by the [Lexilla](https://github.com/ScintillaOrg/lexilla) project.  They are written in C++ for a specific language and tend to not be flexible, after all most languages don't change much.

Notepad++ also allows syntax highlighting with its own styler IIUC (don't and can't use it as it is Windows only), but that is its own code AFAIK and isn't a standard part of Lexilla.  If "somebody" ported that code or maybe a regex based styler to Geany it might be accepted.
